File: python/array/contains_duplicate.py
# 217. Contains Duplicate
# https://leetcode.com/problems/contains-duplicate/description/

from typing import List, Dict, Tuple, Optional

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Create set to store unique elements from the array
        seen_number = set()
        for num in nums:
            # check if the element is already in the hashset
            if num in seen_number:
                return True
            # add element to set
            seen_number.add(num)

        return False # no duplicate





if __name__ == '__main__':
    nums = [1,2,3,1]
    sol = Solution()
    print(sol.containsDuplicate(nums))

File: java/array/contains_duplicate.java
// 217. Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/description/

import java.util.*;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Create hashset to store unique elements from the array
        HashSet<Integer> seen_number = new HashSet<>();

        // iterate through each element in array
        for (int num : nums){
            // check if the element is already in the hashset
            if (seen_number.contains(num)){
                return true;
            }
            // add element to hashset
            seen_number.add(num);
        }
        return false; // no duplicate
    }
}


public class contains_duplicate{
    public static void main(String[] args){
        int[] nums = {1,2,3,1};
        Solution sol = new Solution();
        System.out.println(sol.containsDuplicate(nums));
    }
}
